bin/propolis-server/src/lib/vm/ensure.rs

```diff
@@ -199,7 +199,7 @@ impl<'a> VmEnsureNotStarted<'a> {
         let ps2ctrl = init.initialize_ps2(&chipset)?;
         init.initialize_qemu_debug_port()?;
         init.initialize_qemu_pvpanic(properties.into())?;
-        init.initialize_network_devices(&chipset)?;
+        init.initialize_network_devices(&chipset, &maybe_kstat_sampler).await?;
```

Comment:
Instead of having to pass `maybe_kstat_sampler` as an arg to every init function that needs it, I'd strongly consider making it a `MachineInitializer` member at this point. (I think this would involve removing `MachineInitializer::create_kstat_sampler` and then just calling its callee free function directly from here; this shouldn't be too bad since we have all the MachineInitializer members in this function anyway.)

Reply:
Yeah. I had a variation of this earlier before moving the sampler itself into initializer/ensure, as per @bnaecker's PR. I'll do this.